packages/ethereum/contracts/HoprDistributor.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./HoprToken.sol";
+import "./utils/console.sol";
+
+contract HoprDistributor is Ownable {
+    using SafeMath for uint256;
+
+    // helps us create more accurate calculations
+    uint32 public constant MULTIPLIER = 10 ** 6;
+
+    // total amount minted
+    uint128 public totalMinted = 0;
+
+    // time where the contract will consider as starting time
+    uint32 public startTime;
+    // token which will be used
+    HoprToken public token;
+    // maximum tokens allowed to be minted
+    uint128 public maxMintAmount;
+
+    // A {Schedule} that defined when and how much will be claimed
+    // from an {Allocation}
+    struct Schedule {
+        uint32[] durations;
+        uint32[] percents;
+    }
+
+    // An {Allocation} represents how much a account can claim, claimed,
+    // and when last claim occured
+    struct Allocation {
+        uint128 amount;
+        uint128 claimed;
+        uint32 lastClaim;
+        bool revoked; // account can no longer claim
+    }
+
+    // schedule name -> Schedule
+    mapping(string => Schedule) private schedules;
```

Comment:
why are these mappings private? I always find that annoying for debugging.

Reply:
When a struct contains an array it can't be public, Allocations though are fine, turning them to public.

Reply:
Will create a util function for `schedules` mapping